test/e2e/topology_updater.go

```diff
@@ -239,16 +254,60 @@ var _ = SIGDescribe("Node Feature Discovery topology updater", func() {
 
 	})
 
-	JustAfterEach(func() {
-		err := testutils.DeconfigureRBAC(f.ClientSet, f.Namespace.Name)
-		if err != nil {
-			framework.Logf("failed to delete RBAC resources: %v", err)
-		}
+	When("topology-updater configure to exclude memory", func() {
+		var topologyUpdaterConfigMap *corev1.ConfigMap
+
+		BeforeEach(func() {
+			data := make(map[string]string)
+			data["nfd-topology-updater.conf"] = `excludeList:
+  '*': [memory]
+`
+			topologyUpdaterConfigMap = &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "nfd-topology-updater-conf",
+				},
+				Data: data,
+			}
 
-		err = extClient.ApiextensionsV1().CustomResourceDefinitions().Delete(context.TODO(), crd.Name, metav1.DeleteOptions{})
-		if err != nil {
-			framework.Logf("failed to delete node resources topologies CRD: %v", err)
-		}
+			_, err := f.ClientSet.CoreV1().ConfigMaps(f.Namespace.Name).Create(context.TODO(), topologyUpdaterConfigMap, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			topologyUpdaterDaemonSet, err := f.ClientSet.AppsV1().DaemonSets(f.Namespace.Name).Get(context.TODO(), topologyUpdaterDaemonSet.Name, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			podSpec := &topologyUpdaterDaemonSet.Spec.Template.Spec
+			_ = testutils.NFDTopologyUpdaterConfigMap(podSpec, topologyUpdaterConfigMap.Name, "/etc/kubernetes/node-feature-discovery")
+			topologyUpdaterDaemonSet, err = f.ClientSet.AppsV1().DaemonSets(f.Namespace.Name).Update(context.TODO(), topologyUpdaterDaemonSet, metav1.UpdateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Waiting for daemonset pods to be ready")
+			Expect(testutils.WaitForPodsReady(f.ClientSet, f.Namespace.Name, topologyUpdaterDaemonSet.Spec.Template.Labels["name"], 5)).NotTo(HaveOccurred())
```

Comment:
Hmm, now that we started to talk about end-to-end tests 🙈 I think it would be much cleaner to refactor the creation of the topoogyupdater DaemonSet out of the "main" `BeforeEach()` and directly create it with correct parameters instead of creating it once and then immediately updating it to something else. WDYT?

Reply:
I even thought to take it to the next level and use `JustBeforeEach` in order to decouple configuration from initialization: 
https://onsi.github.io/ginkgo/#separating-creation-and-configuration-justbeforeeach 
I didn't do that in order to keep the commit as minimal as possible.

But now when you raise it sure why not, I'll do it :)

Reply:
> But now when you raise it, sure why not, I'll do it :)

Yeah, go ahead 👍 Sounds really good to me

Reply:
Done!